wagtail/admin/tests/test_userbar.py

```diff
@@ -495,6 +498,54 @@ def construct_wagtail_userbar(request, items):
 
             self.assertTrue(kwargs.get("called"))
 
+    @override_settings(
+        WAGTAIL_I18N_ENABLED=True,
+        WAGTAIL_CONTENT_LANGUAGES=[("en", "English"), ("fr", "French")],
+        WAGTAILADMIN_PERMITTED_LANGUAGES=[("en", "English")],
+        LANGUAGE_CODE="en",
+    )
+    def test_userbar_rendered_in_admin_permitted_language_if_user_has_no_language(self):
+        french = Locale.objects.create(language_code="fr")
+        self.homepage.copy_for_translation(french)
+        french_page = self.page.copy_for_translation(french)
+
+        with translation.override("fr"):
+            response = french_page.serve(self.request)
+            response.render()
+
+        self.assertContains(response, "Go to Wagtail admin")
+
+    @override_settings(
+        WAGTAIL_I18N_ENABLED=True,
+        WAGTAIL_CONTENT_LANGUAGES=[("en", "English"), ("fr", "French")],
+        LANGUAGE_CODE="en",
+    )
+    def test_userbar_rendered_in_active_language_if_admin_permitted(self):
+        french = Locale.objects.create(language_code="fr")
+        self.homepage.copy_for_translation(french)
+        french_page = self.page.copy_for_translation(french)
+
+        with translation.override("fr"):
+            response = french_page.serve(self.request)
+            response.render()
+
+        with translation.override("fr"):
+            expected_text = gettext("Go to Wagtail admin")
```

Comment:
note (non-blocking): I like this and I wish you could make it more explicit. Other than `self.assertNotEqual(expected_text, "Go to Wagtail admin")` I can't think of a nice way to do this.

Reply:
Sure, I added a comment about that